peekingduck/pipeline/nodes/draw/utils/legend.py

```diff
@@ -27,52 +27,65 @@
     FILLED,
     PRIMARY_PALETTE,
     PRIMARY_PALETTE_LENGTH,
-    SMALL_FONTSCALE,
-    THICK,
     WHITE,
 )
 from peekingduck.pipeline.nodes.draw.utils.general import get_image_size
 
+ZONE_COUNTS_HEADING = "-ZONE COUNTS-"
 
-class Legend:
+
+class Legend:  # pylint: disable=too-many-instance-attributes
     """Legend class that uses available info to draw legend box on frame"""
 
-    def __init__(self) -> None:
-        self.legend_left_x = 15
+    def __init__(
+        self,
+        items: List[str],
+        position: str,
+        box_opacity: float,
+        font: Dict[str, Union[float, int]],
+    ) -> None:
+        self.items = items  # list of items to be drawn in legend box
+        self.position = position
+        self.box_opacity = box_opacity
+        self.font_size = font["size"]
+        self.font_thickness = font["thickness"]
 
         self.frame = None
+        self.legend_left_x = 15
         self.legend_starting_y = 0
         self.delta_y = 0
         self.legend_height = 0
+        self.item_height = cv2.getTextSize(
+            "",
+            FONT_HERSHEY_SIMPLEX,
+            self.font_size,
+            self.font_thickness,
+        )[0][1]
+        self.item_padding = self.item_height // 2
 
     def draw(
         self,
         inputs: Dict[str, Any],
-        items: List[str],
-        position: str,
-        box_opacity: float,
     ) -> np.ndarray:
         """Draw legends onto image
 
         Args:
             inputs (dict): dictionary of all available inputs for drawing the legend
-            items (list): list of items to be drawn in legend
-            position (str): used to control whether legend box is drawn on top or bottom
         """
         self.frame = inputs["img"]
 
-        self.legend_height = self._get_legend_height(inputs, items)
-        self.legend_width = self._get_legend_width(inputs, items)
-        self._set_legend_variables(position)
+        self.legend_height = self._get_legend_height(inputs)
+        self.legend_width = self._get_legend_width(inputs)
```

Comment:
Since these are all instance methods/attributes, would it be better refactor these into a `_update_legend_size()` method instead? The `legend_height` and `legend_width` could be updated within the method instead of getting assigned with the return values.

Reply:
Updated